spieleplattform-frontend/src/components/Game/AddGame/AddGame.tsx

```diff
@@ -0,0 +1,125 @@
+import "./AddGame.scss"
+import "react-datepicker/dist/react-datepicker.min.css";
+import DatePicker from "react-datepicker";
+import { useEffect, useState } from "react";
+import GameConsole from "../GameConsole";
+import axios from "axios";
+import { useKeycloak } from "@react-keycloak/web";
+import Error from "../../Error/Error";
+import AddGameDTO from "../AddGameDTO";
+import Game from "../Game";
+
+export default function AddGame() {
+    const [gameConsoles, setGameConsoles] = useState<GameConsole[] | null>(null)
+    const [gameConsoleSelectedFlags, setGameConsoleSelectedFlags] = useState<Boolean[]>(new Array(0))
+    const [errorStatus, setErrorStatus] = useState(null)
+    const { keycloak } = useKeycloak()
+
+    const [name, setName] = useState<string>("")
+    const [releaseDate, setReleaseDate] = useState<Date>(new Date())
+    const [imageUrl, setImageUrl] = useState<string>("")
+    const [youtubeId, setYoutubeId] = useState<string>("")
+    const [description, setDescription] = useState<string>("")
+
+    useEffect(() => {
+        axios.get("/gameConsoles", { headers: { 'Authorization': `Bearer ${keycloak.token}` } })
+            .then((response) => {
+                setGameConsoles(response.data)
+                setGameConsoleSelectedFlags(Array(response.data.length).fill(false))
+            })
+            .catch(error => {
+                setErrorStatus(error.response.status)
+            })
+    }, [keycloak.token])
+
+    function toggleGameConsoleSelectedFlag(index: number) {
+        if (gameConsoleSelectedFlags == null) return
+
+        gameConsoleSelectedFlags[index] = !gameConsoleSelectedFlags[index]
+        setGameConsoleSelectedFlags(structuredClone(gameConsoleSelectedFlags))
+    }
+
+    function submitGame() {
+        if (gameConsoles == null) return
+
+        const gameConsoleIds = new Array<number>(0);
+        for(let index = 0; index < gameConsoles?.length; index++) {
+            if (gameConsoleSelectedFlags[index]) {
+                gameConsoleIds.push(gameConsoles[index].id)
+            }
+        }
+
+        const addGameDTO: AddGameDTO = {
+            name: name,
+            releaseDate: releaseDate,
+            description: description,
+            imageUrl: imageUrl,
+            youtubeId: youtubeId,
+            gameConsoleIds: gameConsoleIds
+        }
+
+        console.log(addGameDTO)
+
+        axios.post<Game>(
+            "/games",
+            addGameDTO,
+            { headers: { 'Authorization': `Bearer ${keycloak.token}` } })
+            .then((game) => {
+                console.log(game.data)
```

Comment:
Normalerweise sollte man keine `console.log()` Statements eingecheckt haben. Kann man natürlich manchmal ganz gut zum Debuggen brauchen aber würde man eigentlich nicht pushen. Alles was du hier loggst, sollte aber auch im Network Tab deiner Browser Developer Tools (also wenn du F12 im Browser drückst) angezeigt werden und zum debuggen in der Regel angenehmer sein.